### ISSUE TITLE
Add e2e test for switching between HelmAddon and ClusterResourceSet strategies

### ISSUE THREAD

**Issue by jimmidyson** [d2iq-labs/capi-runtime-extensions#394]
From @dkoshkin's comment in https://github.com/d2iq-labs/capi-runtime-extensions/pull/393#discussion_r1500946162

              I noticed these Helm labels in other addons, I guess its not a problem for these to be here if not actually managed by helm? I also wonder if there is a helm controller if it tries to "adopt" them.

**Comment by dkoshkin** [comment]
Haha I asked because I was thinking of the scenario where we didn't want this adoption to happen. But this is still a good test to know one way or the other.

**Comment by jimmidyson** [comment]
@dkoshkin Yeah i need to work on https://github.com/kubernetes-sigs/cluster-api/pull/9239 before we can mark variables/fields as immutable.
